app/controllers/users.js

```diff
@@ -15,17 +16,29 @@ const emailIsValid = email => email.match(validEmailPattern);
 
 const passwordIsValid = password => password.match(validPasswordPattern);
 
-exports.createUser = (req, res, next) => {
-  if (!obligatoryParametersWereReceived(req.body)) return next(errors.missingParameters);
+const validateUserCreation = parameters => {
+  if (!obligatoryParametersWereReceived(parameters)) throw errors.missingParameters;
 
-  if (!emailIsValid(req.body.email)) return next(errors.invalidUserEmail);
+  if (!emailIsValid(parameters.email)) throw errors.invalidUserEmail;
+
+  if (!passwordIsValid(parameters.password)) throw errors.invalidUserPassword;
+};
+
+exports.createUser = (req, res, next) => {
+  try {
+    validateUserCreation(req.body);
```

Comment:
dont use a try catch here, either do it with an if and call next directly with the error and have the rest in a promise chain or use async/await and wrap EVERYTHING in the try/catch